Analysis/PIDTools/src/LikelihoodPIDProcessor.cc

```diff
@@ -274,99 +309,63 @@ void LikelihoodPIDProcessor::processEvent( LCEvent * evt ) {
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-    
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID2, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-    //use just Shower Profile variables  (Algorithm 3)
-    _myPID->setBasicFlg(false);
-    _myPID->setdEdxFlg(false);
-    _myPID->setShowerShapesFlg(true);
-    parttype = _myPID->Classification(pp, trk, clu);
-    if(parttype<0) parttype=2;
-    
-    //mu-pi Separation for very low momentum tracks (from 0.2 GeV until 2 GeV)
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-    
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID3, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-    //calculate global likelihood (Algorithm 4)
-    _myPID->setBasicFlg(true);
-    _myPID->setdEdxFlg(true);
-    _myPID->setShowerShapesFlg(true);
-    parttype = _myPID->Classification(pp, trk, clu);
-    if(parttype<0) parttype=2;
-
-    //mu-pi Separation for very low momentum tracks (from 0.2 GeV until 2 GeV)
-    MVAoutput = -100.0;
-    if((parttype==1 || parttype==2) && (_UseMVA && pp.P()<2.0 && clu.size()!=0)){
-      int tmpparttype = _mupiPID->MuPiSeparation(pp, trk, clu);
-      if(_mupiPID->isValid()) parttype = tmpparttype;
-      MVAoutput = _mupiPID->getMVAOutput();   
-    }
-
-    //create PIDHandler
-    createParticleIDClass(parttype, part, pidh, algoID4, MVAoutput);
-
-//////////////////////////////////////////////////////////////////////////////////
-
-    //Low momentum Muon identification  (Algorithm 5)
-    // (from 0.2 GeV until 2 GeV)   
-    if(_UseMVA){
-      MVAoutput = -100.0;
-      parttype = -1;   
-      if(pp.P()<2.5){
-	parttype=_mupiPID->MuPiSeparation(pp, trk, clu);
-	MVAoutput = _mupiPID->getMVAOutput();   
+    for(int ialgo=0; ialgo<_methodstorun.size(); ialgo++) {
+      if(_methodstorun.at(ialgo).compare("LowMomID")==0) {
+	//Low momentum Muon identification 
+	// (from 0.2 GeV until 2 GeV)   
+	if(_UseMVA){
+	  Float_t MVAoutput = -100.0;
+	  parttype = -1;   
+	  if(pp.P()<2.5){
+	    parttype=_mupiPID->MuPiSeparation(pp, trk, clu);
+	    MVAoutput = _mupiPID->getMVAOutput();   
+	  }
+	  //create PIDHandler
+	  createParticleIDClass(parttype, part, pidh, algoID[ialgo], MVAoutput);
+	}
+      } else {
+	//several partivle IDs performed
+	//use just basic variables
+	if(_methodstorun.at(ialgo).compare("BasicVariablePID")==0) {
```

Comment:
```suggestion
	if(_methodstorun.at(ialgo) == "BasicVariablePID") {
```
Not entirely sure whether it is much faster, but I just spent a good 15 minutes trying to understand what `std::string::compare` does and I am still not sure if I do ;). Of course my suggestion is only valid if these should indeed be a 1 to 1 comparison and not one that simply checks if e.g. the name just starts with the desired string. If that is the case then I would prefer the `std::string::find` approach that is also used below.

Reply:
I agree, the simple " == "  solution is probably better since it is a 1 to 1 comparison
I may overthough how to do the comparison.